### PR TITLE
feat(framework) Deprecate `--server`, introduce `--superlink` to connect `ServerApp` and `SuperNode` to `SuperLink`

### PR DESCRIPTION
**Pull request by jafermarq**
Both `flower-client-app` and `flower-server-app` pass the arg `--server` to indicate the address to connect to the `SuperLink`. This PR logs a deprecated warning when doing so and instead encourages users to use two newly introduced arguments:
- `--superlink` as a replacement of `--server` when launching a `ServerApp`.
- `--superlink` as a replacement of `--server` when launching a `SuperNode`

### CODE REVIEW

**Review by danielnugraha — DISMISSED** [review]
Looks good to me. One remark: for new users it's a bit confusing to differentiate Fleet and Driver API. Do we have any documentation about it or should we expose Fleet and Driver APIs to users at all?

**Comment by jafermarq** [comment]
> Looks good to me. One remark: for new users it's a bit confusing to differentiate Fleet and Driver API. Do we have any documentation about it or should we expose Fleet and Driver APIs to users at all?

Fair point! Should we rename it to `--superlink` for both? @danielnugraha @danieljanes

File: src/py/flwr/client/supernode/app.py [hunk 1]
```diff
@@ -63,6 +65,21 @@ def run_client_app() -> None:
...
+        warn = "Passing flag --server is deprecated. Use --superlink-fleet-api instead."
+        warn_deprecated_feature(warn)
+
+        if args.superlink_fleet_api != ADDRESS_FLEET_API_GRPC_RERE:
+            # if `--superlink-fleet-api` also passed, then
+            # warn user that this argument overrides what was passed with `--server`
+            log(
+                WARN,
+                "Both `--server` and `--superlink-fleet-api` were passed. "
+                "`--server` will be ignored.",
```
Comment (danielnugraha) [inline src/py/flwr/client/supernode/app.py:78]: Suggestion: "`--server` will be ignored. Connecting to the Superlink Fleet API at {args.superlink_fleet_api}."

Comment (jafermarq) [reply src/py/flwr/client/supernode/app.py]: I added this.

File: src/py/flwr/server/run_serverapp.py [hunk 2]
```diff
@@ -175,9 +194,14 @@ def _parse_args_run_server_app() -> argparse.ArgumentParser:
...
     parser.add_argument(
         "--server",
-        default="0.0.0.0:9091",
+        default=ADDRESS_DRIVER_API,
         help="Server address",
     )
+    parser.add_argument(
+        "--superlink",
+        default=ADDRESS_DRIVER_API,
+        help="Driver API (gRPC-rere) SuperLink address (IPv4, IPv6, or a domain name)",
```
Comment (danieljanes) [inline src/py/flwr/server/run_serverapp.py:203]:
```suggestion
        help="SuperLink Driver API (gRPC-rere) address (IPv4, IPv6, or a domain name)",
```

File: src/py/flwr/client/supernode/app.py [hunk 3]
```diff
@@ -213,9 +232,14 @@ def _parse_args_common(parser: argparse.ArgumentParser) -> None:
...
     parser.add_argument(
         "--server",
-        default="0.0.0.0:9092",
+        default=ADDRESS_FLEET_API_GRPC_RERE,
         help="Server address",
     )
+    parser.add_argument(
+        "--superlink",
+        default=ADDRESS_FLEET_API_GRPC_RERE,
+        help="Fleet API (gRPC-rere) SuperLink address (IPv4, IPv6, or a domain name)",
```
Comment (danieljanes) [inline src/py/flwr/client/supernode/app.py:241]:
```suggestion
        help="SuperLink Fleet API (gRPC-rere) address (IPv4, IPv6, or a domain name)",
```

**Review by danieljanes — DISMISSED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**